sfge/src/main/java/com/salesforce/graph/build/StaticBlockUtil.java

```diff
@@ -0,0 +1,340 @@
+     *     the child count index and does not necessarily follow sequence
+     * @return new synthetic method vertex with name SyntheticStaticBlock_%d, which will be the
+     *     parent for blockStatementVertex, and a sibling of <clint>()
+     */
+    public static Vertex createSyntheticStaticBlockMethod(
+            GraphTraversalSource g,
+            Vertex clintVertex,
+            Vertex blockStatementVertex,
+            int staticBlockIndex) {
+        final Vertex syntheticMethodVertex = g.addV(ASTConstants.NodeType.METHOD).next();
+        final String definingType = clintVertex.value(Schema.DEFINING_TYPE);
+        addSyntheticStaticBlockMethodProperties(
+                g, definingType, syntheticMethodVertex, staticBlockIndex);
+
+        final Vertex modifierNodeVertex = g.addV(ASTConstants.NodeType.MODIFIER_NODE).next();
+        addStaticModifierProperties(g, definingType, modifierNodeVertex);
+        GremlinVertexUtil.addParentChildRelationship(g, syntheticMethodVertex, modifierNodeVertex);
+
+        GremlinVertexUtil.makeSiblings(g, clintVertex, syntheticMethodVertex);
+
+        return syntheticMethodVertex;
+    }
+
+    /**
+     * If rootNode contains synthetic static block methods (created through {@link
+     * #createSyntheticStaticBlockMethod}), adds a top-level synthetic method that invokes each
+     * static block method.
+     */
+    public static void createSyntheticStaticBlockInvocation(
+            GraphTraversalSource g, Vertex rootNode) {
+        // Check if root node contains any static block methods
+        final List<Vertex> staticBlockMethods = getStaticBlockMethods(g, rootNode);
+
+        // Create static block invocation method
+        if (!staticBlockMethods.isEmpty()) {
+            final String definingType = rootNode.value(Schema.DEFINING_TYPE);
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(
+                        "Creating synthetic invocation method for {} to invoke {} static block(s)",
+                        definingType,
+                        staticBlockMethods.size());
+            }
+
+            final Vertex invokerMethodVertex =
+                    createSyntheticInvocationsMethod(g, rootNode, staticBlockMethods, definingType);
+            MethodPathBuilderVisitor.apply(g, invokerMethodVertex);
+        }
+    }
+
+    static boolean isStaticBlockStatement(JorjeNode node, JorjeNode child) {
+        return isClintMethod(node) && containsStaticBlock(node) && isBlockStatement(child);
+    }
+
+    private static Vertex createSyntheticInvocationsMethod(
+            GraphTraversalSource g,
+            Vertex rootNode,
+            List<Vertex> staticBlockMethods,
+            String definingType) {
+        // Create new synthetic method StaticBlockInvoker to invoke each synthetic static block
+        // method
+        final Vertex invokerMethodVertex = g.addV(ASTConstants.NodeType.METHOD).next();
```

Comment:
Artificially creating a method and its structure to invoke each static block method.